src/item.cpp

```diff
@@ -2373,57 +2395,127 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
     if( item_vars.find( "item_note" ) != item_vars.end() ) {
         //~ %s is an item name. This style is used to denote items with notes.
         return string_format( _( "*%s*" ), ret.str().c_str() );
-    } else {
-        return ret.str();
     }
+
+    return ret.str();
 }
 
 std::string item::display_name( unsigned int quantity ) const
 {
-    std::string name = tname( quantity );
-    std::string sidetxt;
-    std::string amt;
-
-    switch( get_side() ) {
-        case side::BOTH:
-            break;
-        case side::LEFT:
-            sidetxt = string_format( " (%s)", _( "left" ) );
-            break;
-        case side::RIGHT:
-            sidetxt = string_format( " (%s)", _( "right" ) );
-            break;
+    // Anonymous enum used for readability
+    enum {
+        NAMING_ERROR,
+        NAMING_STANDARD,            // Wrench
+        NAMING_WITH_CHARGES,        // 00 Shot (10), Flashlight (100)
+        NAMING_WITH_AMMO_TYPE,      // Glock 18 (15) (9x19mm)
+        NAMING_CONTAINER_LEADING,   // plastic bottle with clean water (2)
+        NAMING_CONTAINER_TRAILING   // clean water (2) in plastic bottle
+    } naming_style ;
+    
+    // Using NAMING_ERROR by default; If this value reaches the end of the
+    // function something has gone wrong.
+    naming_style = NAMING_ERROR;
+    
+    // Allocate all the strings we might need
+    std::string item_text = this->tname( quantity );
+    std::string charges_text = "";
+    std::string content_text = "";
+    std::string ammo_type_text = "";
+
+    // Various checks to determine naming_style and which strings to display
+    unsigned int charges = 0;
+    
+    if ( false ) { 
+        // Dumb hack so it's easier to rearrange code. Remove this when done.
+    } else if ( this->is_container() && !this->contents.empty() ) {
+        naming_style = NAMING_CONTAINER_TRAILING;
```

Comment:
Instead of having an error state, you could move the logic to determine the naming style into a function:
```C++
naming_style make_up_a_function_name( ... ) {
    if( ... ) {
        return NAMING_CONTAINER;
    } else {
        ...
    }
    // compiler will tell you when you forgot to return a value in one code path
}

void your_main_function( ... ) {
    const naming_style style = make_up_a_function_name( ... );
    // And now style won't ever by the error state, and it's const now!
```